plugins/flutter_aepuserprofile/android/src/main/java/com/adobe/marketing/mobile/flutter/flutter_aepuserprofile/FlutterAEPUserProfilePlugin.java

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.flutter.flutter_aepuserprofile;
+
+import android.annotation.SuppressLint;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.adobe.marketing.mobile.AdobeCallbackWithError;
+import com.adobe.marketing.mobile.AdobeError;
+import com.adobe.marketing.mobile.UserProfile;
+
+import org.json.JSONObject;
+
+import java.util.List;
+import java.util.Map;
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+
+
+/**
+ * FlutterAEPUserProfilePlugin
+ */
+public class FlutterAEPUserProfilePlugin implements FlutterPlugin, MethodCallHandler {
+
+  private final String TAG = "FlutterAEPUserProfilePlugin";
+  private MethodChannel channel;
+
+  @Override
+  public void onAttachedToEngine(@NonNull final FlutterPluginBinding binding) {
+    channel = new MethodChannel(binding.getBinaryMessenger(), "flutter_aepuserprofile");
+    channel.setMethodCallHandler(new FlutterAEPUserProfilePlugin());
+  }
+
+  @Override
+  public void onDetachedFromEngine(@NonNull final FlutterPluginBinding binding) {
+    if (channel != null) {
+      channel.setMethodCallHandler(null);
+    }
+  }
+
+  @Override
+  public void onMethodCall(MethodCall call, Result result) {
+    if ("extensionVersion".equals(call.method)) {
+      result.success(UserProfile.extensionVersion());
+    } else if ("getUserAttributes".equals((call.method))) {
+      handleGetUserAttributes(call.arguments, result);
+    } else if ("removeUserAttributes".equals(call.method)) {
+      handleRemoveUserAttributes(call.arguments);
+      result.success(null);
```

Comment:
It's possible for the result to be success even if the arguments are invalid. Is that expected?

Reply:
Dart only allows passing a `List<String>` to this API. This call completes the future as even our native extension API does not return any value to denote if the call has succeeded.